### DISCUSSION TITLE
Is there a way to determine what tags are included/excluded at run time? Or mock based on tags?

### DISCUSSION THREAD

**Discussion by briantist** [pester/Pester#2322] [Q&A] [answered]
The basic idea is that I have some tests that should basically look the same whether I'm implementing them as unit tests or integration tests.

I'd like to not duplicate them, and instead use tags like `Unit` and `Integration`, where the only difference is that if the tag is `Unit` (or `-not` `Integration`) then I will create certain mocks, and otherwise I will not.

Is there a way to do this?

**Comment by fflaten** [comment, marked as answer]
Checking configuration at runtime will depend on https://github.com/pester/Pester/issues/318.
If you don't mind relying on internal API that will likely change (ex. the variable name when made public), then you can use:

```ps1
$____Pester.Configuration.Filter.Tag.Value
$____Pester.Configuration.Filter.ExcludeTag.Value
```

**Supported workaround:**
You can use container/script Data to set a variable, ex.
```ps1
$containers = New-PesterContainer -Path '/workspaces/Pester/Samples/' -Data @{ RunAsUnitTests = $true }
# Set in Run.Container if using Advanced Configuration

Invoke-Pester -Container $containers -Output Detailed
```

`$RunAsUnitTests` would then be available in both Discovery and Run-phase.

If your testsfiles define it as a parameter as well, ex. `param([switch]$RunAsUnitTests)`, you'll also benefit from the default value if forgotten in `-Data`. It would also work with interactive execution.
```ps1
& ./myTestFile.tests.ps1                   # $RunAsUnitTests would be false. Typically F5 in VSCode
& ./myTestFile.tests.ps1 -RunAsUnitTests   # $RunAsUnitTests would be true
```

**Reply by briantist** [reply]
Thank you! Using a parameter was my next move, but I wanted to see if I could leverage tags instead.

The last comment on #318 is 4 years old, so it doesn't really seem like anything is going to happen there soon.

But at least I know that there wasn't some thing I was missing, so I can proceed with other approaches now. Thanks again!
